File: majorProject/src/main/java/com/aocc/majorproject/TutorialScreen.java
package com.aocc.majorproject;

import java.util.List;

import android.graphics.Color;
import android.graphics.Paint;

import com.aocc.framework.Graphics;
import com.aocc.framework.Screen;
import com.aocc.framework.Input.TouchEvent;
import com.aocc.majorproject.ui.UiButton;

public class TutorialScreen extends Screen {

	MajorProjectGame majorProjectGame;
	Paint paint;
	private final UiButton menuButton = new UiButton(0, 0, 200, 100, "Menu");
	
	public TutorialScreen(MajorProjectGame game) {
		super(game);
		majorProjectGame = game;
		// Define a paint object
		paint = new Paint();
        paint.setTextSize(30);
        paint.setTextAlign(Paint.Align.CENTER);
        paint.setAntiAlias(true);
        paint.setColor(Color.WHITE);
	}
	
	@Override
	public void update(float deltaTime) {
		//creates list of all touch events
		
		List<TouchEvent> touchEvents = game.getInput().getTouchEvents();
		
		int len = touchEvents.size();
		
		for (int i = 0; i < len; i++) {
			TouchEvent event = touchEvents.get(i);	// **add error detection
				
			if (event.type == TouchEvent.TOUCH_UP) {
		
				if (menuButton.touchInBounds(event)){
		    		Assets.tap.play(MainMenuScreen.tapVol);
					game.setScreen(new MainMenuScreen(majorProjectGame));
				}
			}
		}
	}
	
	@Override
	public void paint(float deltaTime) {
		Graphics g = game.getGraphics();
		g.drawImage(Assets.tutorial, 0, 0);
		paint.setTypeface(Assets.plain);
		menuButton.paint(g, paint);

        VersionOverlay.paint(g, paint);
	}
	

	@Override
	public void backButton() {
		game.setScreen(new MainMenuScreen(majorProjectGame));
	}

	@Override
	public void pause() {
		// TODO Auto-generated method stub
		
	}

	@Override
	public void resume() {
		// TODO Auto-generated method stub
		
	}

	@Override
	public void dispose() {
		// TODO Auto-generated method stub
		
	}
}

File: majorProject/src/main/kotlin/com/aocc/majorproject/TutorialScreen.kt
package com.aocc.majorproject

import android.graphics.Color
import com.aocc.framework.Graphics
import com.aocc.framework.Screen
import com.aocc.majorproject.input.GamepadInput
import com.aocc.majorproject.ui.UiButton

class TutorialScreen(private val majorProjectGame: MajorProjectGame) : Screen(majorProjectGame) {

    private val menuButton = UiButton.menuAt(0, 0)

    override fun update(deltaTime: Float) {
        val touchEvents = game.input.touchEvents
        for (event in touchEvents) {
            if (event.type == com.aocc.framework.Input.TouchEvent.TOUCH_UP) {
                if (menuButton.touchInBounds(event)) {
                    goToMenu()
                }
            }
        }

        for (action in majorProjectGame.getGamepadInput().consumeActions()) {
            if (action == GamepadInput.Action.CONFIRM || action == GamepadInput.Action.CANCEL) {
                goToMenu()
            }
        }
    }

    private fun goToMenu() {
        Assets.tap?.play(GamePreferences.getTapVolume().toFloat())
        game.setScreen(MainMenuScreen(majorProjectGame))
    }

    override fun paint(deltaTime: Float) {
        val g = game.graphics
        Assets.tutorial?.let { g.drawImage(it, 0, 0) }
        g.drawRect(0, 0, UiButton.MENU_WIDTH, LEGACY_MENU_ART_HEIGHT, Color.BLACK)
        menuButton.paint(g)
        VersionOverlay.paint(g)
    }

    override fun backButton() {
        game.setScreen(MainMenuScreen(majorProjectGame))
    }

    override fun pause() {
    }

    override fun resume() {
    }

    override fun dispose() {
    }

    companion object {
        /** Legacy tutorial art includes a 200x100 menu button graphic beneath our control. */
        private const val LEGACY_MENU_ART_HEIGHT = 100
    }
}
